src/Microsoft.DotNet.GenAPI/Tests/CSharpBuilderTests.cs

```diff
@@ -65,86 +89,501 @@ record struct RecordStruct { }
+            namespace A
+            {
+                public enum Color
+                {
+                    White = 0,
+                    Green = 100,
+                    Blue = 200,
+                }
+            }
+            """;
+
+        var assembly = CompilationHelper.GetAssemblyFromSyntax(syntaxTree, enableNullable: false);
+
+        var builder = new CSharpBuilder(
+            _orderProvider.Object,
+            _assemblySymbolFilter.Object,
+            _syntaxWriter.Object);
+        builder.WriteAssembly(assembly);
+
+        _syntaxWriter.Verify(o => o.WriteTypeDefinition(
+            new[] { SyntaxKind.PublicKeyword },
+            new[] { SyntaxKind.EnumKeyword },
+            "Color",
+            new string[] { },
+            new string[] { }));
+        
+        _syntaxWriter.Verify(o => o.WriteField("White = 0"));
+        _syntaxWriter.Verify(o => o.WriteField("Green = 100"));
+        _syntaxWriter.Verify(o => o.WriteField("Blue = 200"));
+        _syntaxWriter.Verify(o => o.WriteField("White = 0"));
+
+        _syntaxWriter.Verify(o => o.WriteMethod("public Color()", true));
+
+        _syntaxWriter.Verify(o => o.WriteNamespace(new string[] { /* global namespace */ }));
+        _syntaxWriter.Verify(o => o.WriteNamespace(new[] { "A" }));
+
+        _syntaxWriter.VerifyNoOtherCalls();
+    }
+
+    [Fact]
+    public void TestPropertyGeneration()
+    {
+        var syntaxTree = """
+            namespace A
+            {
+                class Car
+                {
+                    int? Wheels { get; }
+                    public bool IsRunning { get; set; }
+                }
+            }
+            """;
+
+        _assemblySymbolFilter.Setup(o => o.Include(It.IsAny<ISymbol>())).Returns(FilterOutImplicitMethods);
+
+        var assembly = CompilationHelper.GetAssemblyFromSyntax(syntaxTree, enableNullable: false);
+
+        var builder = new CSharpBuilder(
+            _orderProvider.Object,
+            _assemblySymbolFilter.Object,
+            _syntaxWriter.Object);
+        builder.WriteAssembly(assembly);
+
+        _syntaxWriter.Verify(o => o.WriteNamespace(new string[] { /* global namespace */ }));
+        _syntaxWriter.Verify(o => o.WriteNamespace(new[] { "A" }));
+
+        _syntaxWriter.Verify(o => o.WriteTypeDefinition(
+            new[] { SyntaxKind.InternalKeyword },
+            new[] { SyntaxKind.PartialKeyword, SyntaxKind.ClassKeyword },
+            "Car",
+            new string[] { },
+            new string[] { }));
 
-        var builder = new CSharpBuilder(orderProvider.Object,
-            assemblySymbolFilter.Object, syntaxWriter.Object);
+        _syntaxWriter.Verify(o => o.WriteProperty(
+            "private int? Wheels",
+            /*hasImplementation*/true,
+            /*hasGet*/true,
+            /*hasSet*/false));
+
+        _syntaxWriter.Verify(o => o.WriteProperty(
+            "public bool IsRunning",
+            /*hasImplementation*/true,
```

Comment:
It might be better to use this:
```suggestion
            hasImplementation: true,
```

Reply:
There's a VS refactoring for this to generate it automatically

Reply:
I tried this, but compilation error occurred. `An expression tree may not contain a named argument specification`